src/custom_de.rs

```diff
@@ -94,31 +104,39 @@ impl<'de> de::Deserializer<'de> for &mut EthFixedDeserializer {
     }
 
     fn deserialize_u32<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
+        if self.remaining_bytes <= 0 {
```

Comment:
`remaining_bytes` is `usize` so cannot be less than zero.  Comparison to 0 is wrong, since we will subtract 4 in L114.  This should be `if self.remaining_bytes < 4 {`.

Reply:
you are so right, let me fix this